### PR TITLE
Converts Nutrition alert to hud element

### PR DESCRIPTION
**Pull request by larentoun**
## What Does This PR Do

Original: https://github.com/ParadiseSS13/Paradise/pull/25555

## What Does This PR Do

Moves nutrition alert to HUD element near a healthdoll. That's basically it.

"Animation" of the alert will be discarded, but it's a minor inconvenience.

## Why It's Good For The Game

Since it's always an alert, why not make it just a hud element?

Also it's required for downstream to not mess up with ui order.

## Images of changes

Screenshot_5

## Testing

Tested with:
- Human
- Machine
- Vampire
- Abductor

## Changelog
:cl:
tweak: Moved Nutrition alert to Nutrition hud element near a healthdoll. It won't run around, when you have multiple alerts.
/:cl:

### CODE REVIEW

**Comment by Coolrune206** [comment]
Why has the nutrition meter migrated? Now it pops out, taking up more of the screen instead of staying in line with the other right-side elements.

**Comment by larentoun** [comment]
@Contrabang @DGamerL I've migrated to here

**Comment by larentoun** [comment]
> Why has the nutrition meter migrated? Now it pops out, taking up more of the screen instead of staying in line with the other right-side elements.

It was taking alert's spot, so there isn't a lot of space left around right-side.

Either top-right (looks kind of stupid and out of place), or bottom-right below stamina. Or just a new place entirely

**Comment by Coolrune206** [comment]
Couldn't go right above or below the health doll?

File: code/__DEFINES/hud_locations.dm [hunk 1]
```diff
@@ -83,30 +83,11 @@
...
-//Upper-middle right (damage indicators)
-#define ui_toxin "EAST-1:28,CENTER+5:27"
-#define ui_fire "EAST-1:28,CENTER+4:25"
-#define ui_oxygen "EAST-1:28,CENTER+3:23"
-#define ui_pressure "EAST-1:28,CENTER+2:21"
-
-#define ui_alien_toxin "EAST-1:28,CENTER+5:25"
-#define ui_alien_fire "EAST-1:28,CENTER+4:25"
-#define ui_alien_oxygen "EAST-1:28,CENTER+3:25"
-#define ui_alien_nightvision "EAST-1:28,CENTER+2:25"
```
Comment (Sirryan2002) [inline code/__DEFINES/hud_locations.dm:102]: is there a reason why you removed all of these defines?

Comment (larentoun) [reply code/__DEFINES/hud_locations.dm:102]: They are not used

Comment (larentoun) [reply code/__DEFINES/hud_locations.dm:102]: Seems to be legacy from... Bay?

**Review by Sirryan2002 — APPROVED** [review]
code looks fine

**Comment by larentoun** [comment]
I would've loved more input about the hud element placement

**Comment by Burzah** [comment]
> I would've loved more input about the hud element placement

Well...
I think the temperature meter should be next to the health doll. It isn't something that is always displayed, only when it needs to be, so that if it is on the side of the health doll only when there is a problem, that would be okay.

After those two things switch places, you could switch the places further between where the no gravity indicator is, and where the new hunger meter location is. So that way the order from the bottom would be health doll, stamina meter, hunger meter, and then gravity. 

But this is just an idea, you don't have to do that, but I think those things work better in the amount they are most likely to occur in a round.

**Comment by larentoun** [comment]
> > I would've loved more input about the hud element placement
> 
> Well... I think the temperature meter should be next to the health doll. It isn't something that is always displayed, only when it needs to be, so that if it is on the side of the health doll only when there is a problem, that would be okay.
> 
> After those two things switch places, you could switch the places further between where the no gravity indicator is, and where the new hunger meter location is. So that way the order from the bottom would be health doll, stamina meter, hunger meter, and then gravity.
> 
> But this is just an idea, you don't have to do that, but I think those things work better in the amount they are most likely to occur in a round.

All the things above the health are called alerts. They don't have a specific place, only these 5 spots. Whatever is more recent gets placed next...

**Comment by Burzah** [comment]
> > > I would've loved more input about the hud element placement
> > 
> > 
> > Well... I think the temperature meter should be next to the health doll. It isn't something that is always displayed, only when it needs to be, so that if it is on the side of the health doll only when there is a problem, that would be okay.
> > After those two things switch places, you could switch the places further between where the no gravity indicator is, and where the new hunger meter location is. So that way the order from the bottom would be health doll, stamina meter, hunger meter, and then gravity.
> > But this is just an idea, you don't have to do that, but I think those things work better in the amount they are most likely to occur in a round.
> 
> All the things above the health are called alerts. They don't have a specific place, only these 5 spots. Whatever is more recent gets placed next...

Yeah that is fair. I am just thinking about how it looks. I have not really looked much into hud element placement. There are probably improvements to be made with it, but would be OOS of this. I think the change is fine overall, I think some people will just find it jarring because of the new placement.

**Review by lewcc — APPROVED** [review]
lgtm, we can always move it later if people are concerned about it

**Comment by Victory1433** [comment]
It would be way nicer if it was in line with the other elements rather than jutting out. Either above or below, but not out to the side.

**Pull request merged**